### PR TITLE
[FIX] Word Cloud: test and prevention empty data

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
WordCloud crashes on empty corpus and empty topic.

##### Description of changes
Do not plot when corpus/topic is empty.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by jerneju** [comment]
https://sentry.io/biolab/orange3-text-bl/issues/275668373/

File: orangecontrib/text/widgets/tests/test_owworldclooud.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        self.send_signal("Corpus", self.corpus)
+        self.send_signal("Corpus", None)
+
+    def test_empty_data(self):
+        """
+        Widget crashes when receives zero length data.
+        GH-244         
+        """
+        corpus = self.corpus
+        corpus.X = np.ones((0, 3))
```
Comment (nikicc) [inline orangecontrib/text/widgets/tests/test_owworldclooud.py]: Couldn't we just take `self.corpus[:0]`?

File: orangecontrib/text/widgets/tests/test_owworldclooud.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    def setUp(self):
+        self.widget = self.create_widget(OWWordCloud)
+        self.corpus = Corpus.from_file('deerwester')
+
+    def test_data(self):
+        """
+        Just basic test.
+        GH-244
+        """
+        self.send_signal("Corpus", self.corpus)
```
Comment (nikicc) [inline orangecontrib/text/widgets/tests/test_owworldclooud.py]: What does this test? If it doesn't crash on data?

File: orangecontrib/text/widgets/owwordcloud.py [hunk 3]
```diff
@@ -235,7 +239,7 @@ def on_corpus_change(self, data):
     def handleNewSignals(self):
         if self.topic is not None:
             self._apply_topic()
-        elif self.corpus is not None:
+        elif self.corpus is not None and len(self.corpus):
```
Comment (kernc) [inline orangecontrib/text/widgets/owwordcloud.py]: Does the same issue exist if _topic_ is empty non-None?

File: orangecontrib/text/widgets/owwordcloud.py [hunk 4]
```diff
@@ -44,6 +44,7 @@ class OWWordCloud(widget.OWWidget):
 
     class Warning(widget.OWWidget.Warning):
         topic_precedence = widget.Msg('Input signal Topic takes priority over Corpus')
+        empty_corpus = widget.Msg('Empty Corpus')
```
Comment (kernc) [inline orangecontrib/text/widgets/owwordcloud.py]: "Input Corpus is empty."

Is there any other widget that does this? @astaric Is this desirable?

Comment (nikicc) [inline orangecontrib/text/widgets/owwordcloud.py]: Perhaps we should just update info box to say: `0 documents with 0 words`?

Comment (jerneju) [inline orangecontrib/text/widgets/owwordcloud.py]: @kernc: Yes, Bag of Words has similar issue.

File: orangecontrib/text/widgets/owwordcloud.py (hunk 3 above)
Comment (nikicc) [inline orangecontrib/text/widgets/owwordcloud.py]: Yup, you're rights. Should be fixed now.

**Pull request merged**